epistemix_platform/src/epistemix_platform/services/results_packager.py

```
@@ -0,0 +1,238 @@
+        with zipfile.ZipFile(zip_buffer, "w", zipfile.ZIP_DEFLATED) as zip_file:
+            for file_path in results_dir.rglob("*"):
+                if file_path.is_file():
+                    arcname = self._calculate_archive_name(file_path, results_dir, run_dirs, is_single_run)
+                    zip_file.write(file_path, arcname=arcname.as_posix())
+                    file_count += 1
+                    logger.debug(f"Added to ZIP: {arcname}")
+
+        return zip_buffer.getvalue(), file_count
+
+    def _calculate_archive_name(
+        self, file_path: Path, results_dir: Path, run_dirs: list[Path], is_single_run: bool
```

Ruff (0.14.1)

Unused method argument: `is_single_run`

(ARG002)